pkgs/development/interpreters/ruby/default.nix

```diff
@@ -197,36 +193,65 @@ let
       }
     ) args; in self;
 
-in {
-  ruby_2_3 = generic {
-    version = rubyVersion "2" "3" "8" "";
+in rec {
+  mkRuby = { version ? null, versionNrs ? null, versionStr ? null, ... } @ args:
```

Comment:
Could also add some documentation to `doc/languages-frameworks/ruby.xml` about this function?